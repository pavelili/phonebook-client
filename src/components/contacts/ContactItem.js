import React from 'react';
import PropTypes from 'prop-types';

export default function ContactItem({ name, phoneNumber }){
  const number = phoneNumber.replace(/(\d{3})(\d{3})(\d{4})/, '$1-$2-$3');
  return(
    <>
      <h3>{name} - Phone #: {number}</h3>
    </>
  );
}

ContactItem.propTypes = {
  name: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired
};
