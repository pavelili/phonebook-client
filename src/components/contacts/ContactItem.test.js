import React from 'react';
import { shallow } from 'enzyme';
import ContactItem from './ContactItem';

describe('ContactItem component', () => {
  it('render a contact item', () => {
    const wrapper = shallow(
      <ContactItem name='George' phoneNumber='1234567890' />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
