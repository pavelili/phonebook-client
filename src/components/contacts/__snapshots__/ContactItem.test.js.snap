// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactItem component render a contact item 1`] = `
<Fragment>
  <h3>
    George
     - Phone #: 
    123-456-7890
  </h3>
</Fragment>
`;
